osfoffline/views/preferences.py

```diff
@@ -164,6 +170,11 @@ def reset_tree_widget(self):
         self.tree_items.clear()
         self.preferences_window.treeWidget.clear()
 
+    def check_tree_changed(self, nodes):
+        if nodes != self.tree_items:
```

Comment:
If I read this correctly, `nodes` is a list of `RemoteNode` objects (see L207) , but `tree_items` is a list of `(tree_item, node.id)` tuples. Are these ever equal?

It seems to me that this doesn't fix the issue directly; it simply ensures that if there are already items in the tree, the cursor is never changed at all.